perception/lidar_transfusion/lib/transfusion_trt.cpp

```diff
@@ -0,0 +1,262 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lidar_transfusion/transfusion_trt.hpp"
+
+#include "lidar_transfusion/preprocess/preprocess_kernel.hpp"
+#include "lidar_transfusion/transfusion_config.hpp"
+
+#include <tier4_autoware_utils/math/constants.hpp>
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace lidar_transfusion
+{
+
+TransfusionTRT::TransfusionTRT(
+  const NetworkParam & network_param, const DensificationParam & densification_param,
+  const TransfusionConfig & config)
+: config_(config)
+{
+  network_trt_ptr_ = std::make_unique<NetworkTRT>(config_);
+
+  network_trt_ptr_->init(
+    network_param.onnx_path(), network_param.engine_path(), network_param.trt_precision());
+  vg_ptr_ = std::make_unique<VoxelGenerator>(densification_param, config_, stream_);
+  stop_watch_ptr_ = std::make_unique<tier4_autoware_utils::StopWatch<std::chrono::milliseconds>>();
+  stop_watch_ptr_->tic("processing/inner");
+  initPtr();
+
+  CHECK_CUDA_ERROR(cudaStreamCreate(&stream_));
+
+#ifdef CPU_PROCESSING
+  RCLCPP_DEBUG_STREAM(rclcpp::get_logger("lidar_transfusion"), "Using CPU processing");
+#else
+  RCLCPP_DEBUG_STREAM(rclcpp::get_logger("lidar_transfusion"), "Using GPU processing");
+#endif
+}
+
+TransfusionTRT::~TransfusionTRT()
+{
+  if (stream_) {
+    cudaStreamSynchronize(stream_);
+    cudaStreamDestroy(stream_);
+  }
+}
+
+void TransfusionTRT::initPtr()
+{
+  // point cloud to voxels
+  voxel_features_size_ = config_.max_voxels_ * config_.max_num_points_per_pillar_ * 5;
```

Comment:
Please do not use magic numbers. You should reuse the variables from the config header